packages/lit-html/src/lit-html.ts

```diff
@@ -920,9 +920,17 @@ class ChildPartImpl {
   _$setValue(value: unknown, directiveParent: DirectiveParent = this): void {
     value = resolveDirective(this, value, directiveParent);
     if (isPrimitive(value)) {
-      if (value === nothing) {
+      // Non-rendering child values. It's important that these do not render
+      // empty text nodes to avoid issues with preventing default <slot>
+      // fallback content.
+      if (
+        value === nothing ||
```

Comment:
I'd do just `value === nothing || value == null`.

I'm a bit uneasy with this behavior for empty strings. If someone specifically put a string there, I'd thing we'd render the string, even if empty.

Reply:
I suppose a footgun to consider is that it's _shorter_ to just write 
```html
<ce-with-slot>${user ? user.name : ''}</ce-with-slot>
```
than any of the alternatives for the else case (optional chaining aside), and so I think that's still going to require the footnote in the docs that that wouldn't trigger the fallback content, which is the footnote Steve is trying to remove with this change.

Reply:
Based on offline chat, we'll keep including empty string to preserve compatibility with SSR, but we'll add dirty checking so we don't always call clear.